tests/integration/test_generate_oci_image.py

```diff
@@ -0,0 +1,643 @@
+            name="golang.org/x/crypto",
+            version="v0.14.0",
+            supplier=SpdxNoAssertion(),
+            download_location=SpdxNoAssertion(),
+            files_analyzed=False,
+            checksums=[
+                Checksum(
+                    algorithm=ChecksumAlgorithm.SHA256,
+                    value="9876543210abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
+                )
+            ],
+            license_concluded=SpdxNoAssertion(),
+            license_declared=SpdxNoAssertion(),
+            copyright_text=SpdxNoAssertion(),
+            external_references=[
+                ExternalPackageRef(
+                    category=ExternalPackageRefCategory.PACKAGE_MANAGER,
+                    reference_type="purl",
+                    locator="pkg:golang/golang.org/x/crypto@v0.14.0",
+                )
+            ],
+        )
+    ]
+
+
+def _get_component_packages() -> list[Package]:
+    """
+    Get a list of packages that will be added only to the component SBOM with a
+    CONTAINS relationship. After contextualization, all of these packages in
+    the final component SBOM should have the relationships still pointing to
+    the component.
+    """
+    return [
+        Package(
+            spdx_id="SPDXRef-Package-component-specific-golang-crypto",
+            name="golang.org/x/stdlib",
+            version="v0.14.0",
+            supplier=SpdxNoAssertion(),
+            download_location=SpdxNoAssertion(),
+            files_analyzed=False,
+            checksums=[
+                Checksum(
+                    algorithm=ChecksumAlgorithm.SHA256,
+                    value="9876543210abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
+                )
+            ],
+            license_concluded=SpdxNoAssertion(),
+            license_declared=SpdxNoAssertion(),
+            copyright_text=SpdxNoAssertion(),
+            external_references=[
+                ExternalPackageRef(
+                    category=ExternalPackageRefCategory.PACKAGE_MANAGER,
+                    reference_type="purl",
+                    locator="pkg:golang/golang.org/x/stdlib@v0.14.0",
+                )
+            ],
+        )
+    ]
+
+
+def _add_packages_with_relationships(sbom: InputSBOM, packages: list[Package]) -> None:
+    """
+    Add packages to an SBOM with CONTAINS relationships to the root package.
+    """
+    for package in packages:
+        relationship = Relationship(
+            spdx_element_id=sbom.root_package_spdx_id,
+            relationship_type=RelationshipType.CONTAINS,
+            related_spdx_element_id=package.spdx_id,
+        )
+        sbom.document.packages.append(package)
+        sbom.document.relationships.append(relationship)
+
+
+def _create_input_sbom(
+    *,
+    creation_info_name: str,
+    spdx_id: str,
+    package_name: str,
+    version: str,
+    checksum_value: str,
+    purl_locator: str,
```

Comment:
It would be great for future improvements to have these args documented in a docstring.